### ISSUE TITLE
Chromium-based extensions can't communicate with desktop programs (KDE Neon, KDE Plasma)

### ISSUE THREAD

**Issue by AmethystEdits** [brave/brave-browser#17537]
## Description 
Extensions that need to communicate with a desktop application to do their job (currently I have two extensions that do this, Plasma integration and uget integration) can't communicate with their desktop applications. This bug appears to be brave specific as Plasma integration's firefox version works fine.
(KDE Neon is essentially Ubuntu with Plasma preinstalled and a bunch of other changes, shouldn't make too much of a difference but it has before)

## Steps to Reproduce

   1. Get any extension that needs to talk with a desktop application to work properly
   2. Get the extension's web application

## Actual result:
Extensions don't seem to be able to talk with the applications, screenshots if helpful
![image](https://user-images.githubusercontent.com/57216385/129812739-26e2e303-9dc3-4ece-8894-c5e4170ac1e8.png)
![image](https://user-images.githubusercontent.com/57216385/129812760-51ea627a-3284-4d86-934c-8c60470aba24.png)

## Expected result:
Extensions working normally, and doing what they're meant to

## Reproduces how often: 

Always does this

## Brave version (brave://version info)

1.28.105 Chromium: 92.0.4515.131 (Official Build) (64-bit)
6b8d6c56ce21e38a72f7c4becb5abc1fa5134f29-refs/branch-heads/4515@{#1933}
Linux

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel?  Haven't checked, wouldn't expect it to as the bug's been around for a while
- Can you reproduce this issue with the nightly channel? Haven't checked, wouldn't expect it to as the bug's been around for a while

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No, still doesn't work
- Does the issue resolve itself when disabling Brave Rewards? No, still doesn't work
- Is the issue reproducible on the latest version of Chrome? No, extensions work fine

## Miscellaneous Information:

In brave://flags, #brave-extension-network-blocking is set to "default"

**Comment by rebron** [comment]
cc: @fmarier

**Comment by fmarier** [comment]
I've never tried this on KDE, but I have successfully used the [GNOME equivalent](https://chrome.google.com/webstore/detail/gnome-shell-integration/gphhapmejobijbbhgpjhcjognlahblep/related) and it requires the [`chrome-gnome-shell` package](https://wiki.gnome.org/Projects/GnomeShellIntegrationForChrome/Installation) to be installed.

Do you have the `plasma-browser-integration` package installed?

I see that you are using the snap package. Are you able to test using the packages in our [official PPA](https://brave.com/linux/#debian-9-ubuntu-1604-and-mint-18)? I'm wondering whether this problem is only in the snap version of Brave, which is subject to extra restrictions.

**Comment by AmethystEdits** [comment]
Yea the integration package is installed, the extension worked fine in the official PPA version so the snap one's broken.

**Comment by fmarier** [comment]
Thanks @AmethystEdits .

@wknapik That might be another snap permission we need to request from Canonical.

**Comment by wknapik** [comment]
Chromium vs. Brave:
```console
% diff -u <(snap connections chromium|awk '{print $3}'|sed 's/chromium//g'|grep -v '^-$'|sort) <(snap connections brave|awk '{print $3}'|sed 's/brave//g'|grep -v '^-$'|sort)|grep '^-[^-]'         
-:audio-record
-:bluez
-:joystick
-:network-bind
-:personal-files
-:system-files
-:system-packages-doc
-gnome-3-28-1804:gnome-3-28-1804
%
```

Firefox vs. Brave:
```console
% diff -u <(snap connections firefox|awk '{print $3}'|sed 's/chromium//g'|grep -v '^-$'|sort) <(snap connections brave|awk '{print $3}'|sed 's/brave//g'|grep -v '^-$'|sort)|grep '^-[^-]'
-:avahi-observe
-:hardware-observe
-firefox:dbus-daemon
-firefox:mpris
-gnome-3-34-1804:gnome-3-34-1804
%
```

Do these extensions work with the Chromium snap? @fmarier do I understand correctly that it does for you? If so, then the only thing that both Chromium and Firefox have that Brave doesn't is the gnome plug, so adding `gnome-3-34-1804:gnome-3-34-1804` sounds like the solution, at least for gnome.

Some instructions for testing this would be helpful. I don't quite get what these extensions are supposed to do.

**Comment by fmarier** [comment]
> do I understand correctly that the gnome extension works for you in the Brave snap?

Not in the snap. I've only tested this in the normal package.

> Some instructions for testing this would be helpful. I don't quite get what these extensions are supposed to do.

Here's what I did:

1. Install the `brave-browser` package from our repo (I only tested the release channel).
2. Install the `chrome-gnome-shell` package from Ubuntu repo.
3. Start Brave and install https://chrome.google.com/webstore/detail/gnome-shell-integration/gphhapmejobijbbhgpjhcjognlahblep
4. Open https://extensions.gnome.org/extension/6/applications-menu/ and click the toggle button:
![gnome-extension-toggle](https://user-images.githubusercontent.com/167821/135696606-d135f3d1-879f-49c1-a0b1-1cf9a23233cd.png)
5. Check that the GNOME extension has loaded and the Applications menu is present in the top-left corner:
![applications-menu](https://user-images.githubusercontent.com/167821/135696698-ed09e019-8c8a-42c4-bac0-22c79ad36cb8.png)

**Comment by soumyatheman** [comment]
This thing is already solved in the firefox snap, so I guess other browsers can also improve from it. New updates have also improved the startup speed in the firefox snap. I think a small team can work on that.

**Comment by soumyaDghosh** [comment]
I think, at this moment brave should enable native messaging also, just like in `firefox`. You guys should use something like this.
https://github.com/canonical/firefox-snap/blob/stable/patches/native-messaging-portal.patch

**Comment by bsclifton** [comment]
@soumyaDghosh do you know if this is working now?  It should have been addressed with https://github.com/brave/brave-browser-snap/pull/25

**Comment by soumyaDghosh** [comment]
Installed `brave` from latest/edge and it's still the same. It can't be fixed by just some changes in the snap manifest. It needs changes in the source.

**Comment by mihaiplesa** [comment]
@soumyaDghosh is this still an issue?

**Comment by soumyaDghosh** [comment]
> [@soumyaDghosh](https://github.com/soumyaDghosh) is this still an issue?

Yes, this is still an issue, and as explained before, this needs fixes in the source code of brave itself. I have never heard any chromium based browsers to make this work from within sandbox, be it snap or flatpak.

**Comment by tr1xem** [comment]
Its native host issue happening on all browser

**Comment by tr1xem** [comment]
u have to do smth like this

sudo ln -s /etc/chromium/native-messaging-hosts /etc/thorium/native-messaging-hosts

**Comment by glocalglocal** [comment]
This is still an issue with Brave flatpak. Firefox and Ungoogled Chromium flatpaks work fine. So, Chromium based browsers can resolve this.

**Comment by guest271314** [comment]
@soumyaDghosh 

> I think, at this moment brave should enable native messaging also, just like in firefox. You guys should use something like this.

Am I reading what you wrote correctly? Native Messaging is not enabled by default in Brave? 

Or, is this purely a Snap issue?

**Comment by soumyaDghosh** [comment]
Chromium architecture is different from Gecko. So it's easy probably to enable native messaging within snap sandbox in firefox but not in chromium. It's sandbox problem not specifically snap problem. Flatpak also has same issue

**Comment by guest271314** [comment]
I don't use Snap or Flatpak. Does Brave somehow use them?
